### PR TITLE
remove codecov

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

**Comment by reaperhulk** [comment]
Merging this will require removing the codecov required check 😄

File: .github/actions/upload-coverage/action.yml [hunk 1]
```diff
@@ -23,3 +23,8 @@ runs:
         name: coverage-${{ steps.coverage-uuid.outputs.COVERAGE_UUID }}
         path: ".coverage.*"
         if-no-files-found: ignore
+    - uses: actions/upload-artifact@v3.0.0
+      with:
+        name: rust-coverage-${{ steps.coverage-uuid.outputs.COVERAGE_UUID }}
+        path: "*.lcov"
+        if-no-files-found: ignore
```
Comment (alex) [inline .github/actions/upload-coverage/action.yml:30]: newline at the end of file

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -321,36 +321,28 @@ jobs:
           LLVM_PROFILE_FILE: "rust-cov/cov-%m-%p.profraw"
       - name: Process coverage data
         run: |
+            set -xe
             cd src/rust/
             cargo profdata -- merge -sparse ../../rust-cov/*.profraw -o pytest-rust-cov.profdata
             cargo profdata -- merge -sparse rust-cov/*.profraw -o cargo-test-rust-cov.profdata
+            COV_UUID=$(python3 -c "import uuid; print(uuid.uuid4())")
+            COV_UUID2=$(python3 -c "import uuid; print(uuid.uuid4())")
```
Comment (alex) [inline .github/workflows/ci.yml:329]: Any reason not to use the same UUID + two prefixes?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -321,36 +321,28 @@ jobs:
...
-          curl -o codecov.sh -f https://codecov.io/bash || \
-            curl -o codecov.sh -f https://codecov.io/bash || \
-            curl -o codecov.sh -f https://codecov.io/bash
-          bash codecov.sh -n "Rust Coverage" -f '*.lcov' -X coveragepy
+              --ignore-filename-regex='/.rustup/toolchains/' --format=lcov > ../../${COV_UUID2}.lcov
 
+            sed -E -i 's/SF:(.*)\/src\/rust\/(.*)/SF:src\/rust\/\2/g' ../../*.lcov
       - uses: ./.github/actions/upload-coverage
-        with:
-          name: "Rust Coverage"
```
Comment (alex) [inline .github/workflows/ci.yml:353]: Why is name removed?

**Review by alex — COMMENTED** [review]
please do a commit with less than 100% coverage and then revert (don't force push over it!)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -321,37 +321,32 @@ jobs:
...
-          bash codecov.sh -n "Rust Coverage" -f '*.lcov' -X coveragepy
+              --ignore-filename-regex='/.rustup/toolchains/' --format=lcov > ../../${COV_UUID}-2.lcov
 
+            sed -E -i 's/SF:(.*)\/src\/rust\/(.*)/SF:src\/rust\/\2/g' ../../*.lcov
       - uses: ./.github/actions/upload-coverage
         with:
           name: "Rust Coverage"
 
+
+
```
Comment (alex) [inline .github/workflows/ci.yml:349]:
```suggestion
```

**Review by alex — APPROVED** [review]

**Pull request merged**